submitter/base.py

```diff
@@ -133,8 +133,7 @@ def __convert_entity(self, entity: Entity, accession: str = None, other_attribut
         return converted_entity
 
     def __process_response(self, response: ArchiveResponse) -> Tuple[str, ArchiveResponse]:
-        archive_response = response.data
-        if 'error_messages' in archive_response:
+        if response.error_messages:
```

Comment:
Perhaps this is already covered by the  ArchiveResponse class...

... but if archive_response is backed by a dictionary which does not contain an `error_messages` attribute, this line will throw:
`AttributeError: 'dict' object has no attribute 'error_messages'`

if you want to check if `response.error_messages` is falsy I think a safer block would be:
```
if 'error_messages' in archive_response && archive_response.error_messages:
```
or more concisely
```
if archive_response.get('error_messages', []):
```

But if ArchiveResponse is not extending 'dict' this is probably catered for

Reply:
@MightyAx `ArchiveResponse` is not a dictionary, it is an object that contains an `error_messages` field.